generated/nidigital/session.py

```diff
@@ -2750,6 +2750,48 @@ def self_test(self):
             raise errors.SelfTestError(code, msg)
         return None
 
+    @ivi_synchronized
+    def write_source_waveform_site_unique(self, waveform_name, waveform_data):
+        '''write_source_waveform_site_unique
+
+        TBD
+
+        Args:
+            waveform_name (str):
+
+            waveform_data ({ site: data, site: data, ... }): Dictionary where each key is the site number and the value is array.array of unsigned int
+
+        '''
+        if len(waveform_data) == 0:
```

Comment:
Is this what the C library does? (as opposed ot an error)
I would have expected an error. Please confirm.

Reply:
C entry point includes 2 size-related parameters - `numWaveforms` and `samplesPerWaveform`.

If `numWaveforms` is 0: `NIDIGITAL_ERROR_INCONSISTENT_NUM_WAVEFORMS_AND_SITES` (The number of waveforms in the 2D array provided to niDigital Write Source Waveform (Site Unique U32) does not match the number of sites specified.) error is thrown.
If `samplesPerWaveform` is 0: `ivierrors::ERR_INVALID_VALUE` is thrown.

Reply:
I cannot use `next(iter(waveform_data))` on an empty list/array.

I can put is special code to just return early, or put in special code to pass in 0 to the driver and let it do what ever it is going to do, which looks like throw an exception.

Reply:
I think `put in special code to pass in 0 to the driver and let it do what ever it is going to do` will be the most consistent with the other ADEs and also return the richest error.

Reply:
Done